chat-server/src/app.ts

```diff
@@ -71,7 +37,41 @@ const reqHandler: RequestHandler = (req, _res, next) => {
   next();
 };
 
-export const setupApp = async () => {
+export const makeApp = async (): Promise<
+  Express & { close(): Promise<void> }
+> => {
+  const {
+    MONGODB_CONNECTION_URI,
+    MONGODB_DATABASE_NAME,
+    VECTOR_SEARCH_INDEX_NAME,
+    OPENAI_ENDPOINT,
+    OPENAI_API_KEY,
+    OPENAI_EMBEDDING_DEPLOYMENT,
+    OPENAI_EMBEDDING_MODEL_VERSION,
+  } = assertEnvVars(CORE_ENV_VARS);
+
+  // Create instances of services
+  const mongodb = new MongoDB(
```

Comment:
in future PR we're killing this `MongoDB` as a self defined type to just use the DB